### PR TITLE
chore(ci): add a workflow that executes e2e tests

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Introduces E2E test workflow to be run on main branch merge and on demand.
Update: It can also run different fork/repoName/branch using GHA workflow inputs.
### What issues does this PR fix or reference?
#210

### CODE REVIEW

**Comment by odockal** [comment]
tested on my fork: https://github.com/odockal/podman-desktop-extension-bootc/actions/runs/8363989131/job/22898241179

**Review by benoitf — COMMENTED** [review]
hi, why is it not on PR check ?


It looks like it's too late after merge

**Comment by odockal** [comment]
@benoitf If it works, I will add as PR check as well.

**Review by cbr7 — APPROVED** [review]

**Comment by odockal** [comment]
> hi, why is it not on PR check ?
> 
> It looks like it's too late after merge

@benoitf We will have more. This is not excluding any other automation beyond merge event. 

I would like to say that on PR check the workflow will behave differently. We should build the extension locally on the workflow, and then use this local image to be installed by the tests. For me this is different effort. Issue here: https://github.com/containers/podman-desktop-extension-bootc/issues/213.

File: .github/workflows/e2e-main.yaml [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#
+# Copyright (C) 2023 Red Hat, Inc.
```
Comment (benoitf) [inline .github/workflows/e2e-main.yaml:2]:
```suggestion
# Copyright (C) 2024 Red Hat, Inc.
```

File: .github/workflows/e2e-main.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  e2e-tests:
+    name: Run E2E tests
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event.inputs.organization }}/${{ github.event.inputs.repositoryName }}
+          ref: ${{ github.event.inputs.branch }}
+          path: ${{ github.event.inputs.repositoryName }}
+        if: github.event_name == 'workflow_dispatch'
```
Comment (benoitf) [inline .github/workflows/e2e-main.yaml:52]: for readability, I would move the if as first line

File: .github/workflows/e2e-main.yaml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+      - uses: actions/checkout@v4
+        with: 
+          path: podman-desktop-extension-bootc
+        if: github.event_name == 'push'
```
Comment (benoitf) [inline .github/workflows/e2e-main.yaml:57]: move it as first line

File: .github/workflows/e2e-main.yaml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+      # Checkout podman desktop
+      - uses: actions/checkout@v4
+        with:
+          repository: containers/podman-desktop
+          ref: main
+          path: podman-desktop
```
Comment (benoitf) [inline .github/workflows/e2e-main.yaml:64]: why do we need to checkout podman desktop ?

Comment (odockal) [reply .github/workflows/e2e-main.yaml:64]: Yes. We need to build repo locally using `yarn test:e2e:build` which build the app in development mode, and then we are spinning up the podman desktop app from there. 
I have not found a way to simply point tests to a podman desktop electron app (binary) and make it run with playwright.

File: .github/workflows/e2e-main.yaml (hunk 2 above)
Comment (odockal) [reply .github/workflows/e2e-main.yaml:52]: that is right. thx.

File: .github/workflows/e2e-main.yaml (hunk 4 above)
Comment (benoitf) [reply .github/workflows/e2e-main.yaml:64]: ok b/c yes I think it's not so nice

**Review by benoitf — APPROVED** [review]

**Review by cdrage — APPROVED** [review]

**Pull request merged**